### ISSUE TITLE
RFE: sbt "scalafmtOnly <file>" command

### ISSUE THREAD

**Issue by fommil** [scalameta/scalafmt#873]
this would be good for ensime integration (I may write this myself at some point if I find I need it)

**Comment by pjrt** [comment]
What is that command doing? Is it meant to only run a single file?

If so, you can do that via `scalafmt -i -f <file>`.

**Comment by olafurpg** [comment]
We could add a command alias or an input task (for autocomplete) that runs `scalafmt -i -f <file>`.

**Comment by pjrt** [comment]
I don't see the point of adding that if `scalafmt -i -f` works. Though a couple of changes that we could make are:

1. `-i` should be implied if `-f` is passed. I don't see the purpose of having the option to print this to stdout
2. We could do away with the whole interface and have scalafmt ALWAYS take a `<file>`. And if that file happens to be a dir, then search recursively. To replicate the current behaviour would mean doing `scalafmt .`

But this is all wishlist. `scalafmt -i -f` works fine atm. Maybe another bulletpoint for v1.0?

**Comment by olafurpg** [comment]
@pjrt

1.  I've kept `-i` since I prefer the user to be explicit if they want to write changes to disk. However, when thinking about it, the `scalafmt` (no args) command doesn't really follow that principle so maybe it should change.

2. iirc, scopt doesn't support trailing arguments, that's why the `-f` is required. Scopt doesn't support space separated lists either, that's why files are comma separated (always bugged me I can't do `scalafmt *.scala`). I've thought about switching to case-app to support both of those.

We should probably move this discussion however to another issue.

**Comment by fommil** [comment]
if there was an alias, then these kinds of changes could be done seemlessley between versions :wink:

**Comment by olafurpg** [comment]
The cli is the most used scalafmt integration so we wouldn't break the flags so easily, in this case it'd be easy to support the old api in parallel with the new one.

**Comment by pjrt** [comment]
@olafurpg scopts does support list of file arguments. This is from their Readme:

```
    arg[File]("<file>...").unbounded().optional().action( (x, c) =>
    c.copy(files = c.files :+ x) ).text("optional unbounded args")
```

Of course, it wouldn't be optional for scalafmt. I'm doing this in my (yet to be released) Scala CTags generator, where I use the interface I described (always require at least a file, if it is a dir, then search recusebly).

**Comment by olafurpg** [comment]
@pjrt Cool! I didn't know about unbounded. Please open an issue, I'd love to make the `-f` flag optional.

**Comment by pjrt** [comment]
Done: https://github.com/scalameta/scalafmt/issues/875

So back to the issue at hand, @fommil does `scalafmt -i -f <file>` work for you?

**Comment by fommil** [comment]
@pjrt yes it does :grinning: I think we can close this unless you want to add an alias.
